src/Microsoft.Framework.PackageManager/Restore/RestoreCommand.cs

```diff
@@ -136,6 +145,138 @@ public RestoreCommand(IApplicationEnvironment env)
             }
         }
 
+        private async Task<bool> RestoreForInstall(string packagesDirectory)
+        {
+            var success = true;
+
+            var sw = new Stopwatch();
+            sw.Start();
+
+            var restoreOperations = new RestoreOperations(Reports.Verbose);
+            var projectProviders = new List<IWalkProvider>();
+            var localProviders = new List<IWalkProvider>();
+            var remoteProviders = new List<IWalkProvider>();
+
+            localProviders.Add(
+                new LocalWalkProvider(
+                    new NuGetDependencyResolver(
+                        packagesDirectory)));
+
+            var effectiveSources = PackageSourceUtils.GetEffectivePackageSources(
+                SourceProvider,
+                FeedOptions.Sources,
+                FeedOptions.FallbackSources);
+
+            AddRemoteProvidersFromSources(remoteProviders, effectiveSources);
+
+            var restoreContext = new RestoreContext()
+            {
+                FrameworkName = ApplicationEnvironment.RuntimeFramework,
+                ProjectLibraryProviders = projectProviders,
+                LocalLibraryProviders = localProviders,
+                RemoteLibraryProviders = remoteProviders
+            };
+
+            if (RestorePackageVersion == null)
+            {
+                var packageFeeds = new List<IPackageFeed>();
+
+                foreach (var source in effectiveSources)
+                {
+                    var feed = PackageSourceUtils.CreatePackageFeed(
+                        source,
+                        FeedOptions.NoCache,
+                        FeedOptions.IgnoreFailedSources,
+                        Reports);
+                    if (feed != null)
+                    {
+                        packageFeeds.Add(feed);
+                    }
+                }
+
+                var package = await PackageSourceUtils.FindLatestPackage(packageFeeds, RestorePackageId);
+
+                if (package == null)
+                {
+                    Reports.Error.WriteLine("Unable to locate the package {0}".Red(), RestorePackageId);
+                    return false;
+                }
+                RestorePackageVersion = package.Version.ToString();
+            }
+
+            Reports.Information.WriteLine("Installing package {0} {1}", RestorePackageId.Bold(), RestorePackageVersion.Bold());
+
+            var graphNode = await restoreOperations.CreateGraphNode(
```

Comment:
Talked offline and refactoring this is too time consuming to make sense. The code is very similar but extremely different. I tried to refactor this initially and ended up with a method with 8 arguments...